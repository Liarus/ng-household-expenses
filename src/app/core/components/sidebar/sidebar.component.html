<mat-sidenav-container fxLayout="column" fxFlexFill>
    <mat-sidenav fxLayout="column"
      mode="side"
      [opened]="true"
    >
      <mat-nav-list>
        <a mat-list-item [routerLink]="item.url"
        *ngFor="let item of menuItems">
          <mat-icon mat-list-icon>{{item.icon}}</mat-icon>
          <p matLine *ngIf="isExpanded">{{item.title}}</p>
        </a>
      </mat-nav-list>
    </mat-sidenav>
    <mat-sidenav-content
      [ngClass]="(isExpanded==true) ? 'app-sidenav-content-expanded' : 'app-sidenav-content-compressed'"
    >
      <ng-content></ng-content>
    </mat-sidenav-content>
</mat-sidenav-container>