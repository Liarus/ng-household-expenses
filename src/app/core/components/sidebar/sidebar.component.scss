.app-sidenav-content-compressed {
  margin-left: 65px !important;
}

.app-sidenav-content-expanded {
  margin-left: 165px !important;
}