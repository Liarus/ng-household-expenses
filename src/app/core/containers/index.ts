export * from './app';
export * from './page-not-found';
