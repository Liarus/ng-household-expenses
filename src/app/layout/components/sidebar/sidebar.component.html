<mat-sidenav-container fxLayout="column" fxFlexFill>
    <mat-sidenav fxLayout="column"
      mode="side"
      [opened]="true"
    >
      <mat-nav-list class="app-nav-list"
        [ngClass]="(isExpanded === true) ? 'app-nav-list-expanded ' : 'app-nav-list-collapsed'"
      >
        <a mat-list-item [routerLink]="item.url"
        *ngFor="let item of menuItems">
          <mat-icon mat-list-icon>{{ item.icon }}</mat-icon>
          <p matLine>{{ item.title }}</p>
        </a>
      </mat-nav-list>
    </mat-sidenav>
    <mat-sidenav-content class="app-sidenav"
      [ngClass]="(isExpanded === true) ? 'app-sidenav-content-expanded' : 'app-sidenav-content-collapsed'"
    >
      <ng-content></ng-content>
    </mat-sidenav-content>
</mat-sidenav-container>