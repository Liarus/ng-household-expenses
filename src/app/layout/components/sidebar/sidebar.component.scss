.mat-sidenav-container{
  min-height: 93vh !important;
  height: 93vh !important;
}

.app-sidenav {
  -webkit-transition: all 0.25s;
  transition: all 0.25s ease;
}

.app-sidenav-content-collapsed {
  margin-left: 65px !important;
}

.app-sidenav-content-expanded {
  margin-left: 165px !important;
}

.app-sidenav-container {
  -webkit-transition: all 0.25s;
  transition: all 0.25s ease;
}

.app-nav-list {
  overflow-x: hidden;
  -webkit-transition: all 0.25s;
  transition: all 0.25s ease;
}

.app-nav-list-expanded {
  width: 165px !important;
}

.app-nav-list-collapsed {
  width: 65px !important;
}