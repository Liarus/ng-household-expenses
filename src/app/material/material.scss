.mat-bottom-sheet-container {
  padding: 0 !important;
}

.mat-table {
  mat-row, mat-header-row, mat-footer-row {
    padding-left: 24px;
    padding-right: 24px;
  }
  
  mat-cell:first-child, mat-footer-cell:first-child, mat-header-cell:first-child {
    padding-left: 0;
  }
  
  mat-cell:last-child, mat-footer-cell:last-child, mat-header-cell:last-child {
    padding-right: 0;
  }
}

mat-expansion-panel {
  mat-expansion-panel-header {
    height: 48px; // height may be different for you
  }
  .mat-expansion-panel-content {
    height: 0;
  }
  &.mat-expanded {
    mat-expansion-panel-header {
      height: 64px; // height may be different for you
    }
    .mat-expansion-panel-content {
      height: auto;
    }
  }
}