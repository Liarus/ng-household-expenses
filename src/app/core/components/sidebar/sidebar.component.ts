import { Component, Input, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-sidebar',
  changeDetection: ChangeDetectionStrategy.OnPush,
  templateUrl: './sidebar.component.html',
  styleUrls: ['./sidebar.component.scss']
})
export class SidebarComponent {
  @Input() isExpanded: boolean;
}
