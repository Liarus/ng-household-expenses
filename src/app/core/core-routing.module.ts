import { Routes, RouterModule } from '@angular/router';

import { LayoutComponent } from './containers';
import { NgModule } from '@angular/core';

export const routes: Routes = [
    {
        path: 'root',
        component: LayoutComponent
    }
];

@NgModule({
    imports: [RouterModule.forRoot(routes)],
    exports: [RouterModule]
  })
export class CoreRoutingModule { }
