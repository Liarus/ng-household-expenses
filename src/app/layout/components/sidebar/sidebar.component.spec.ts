/// <reference types="jest" />
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { ChangeDetectionStrategy } from '@angular/core';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

import { SidebarComponent } from './sidebar.component';
import { MaterialModule } from '../../../material/material.module';
import { MenuItem } from '../../models/menuItem.model';

describe('SidebarComponent', () => {
  let component: SidebarComponent;
  let fixture: ComponentFixture<SidebarComponent>;

  const menuItems: MenuItem[] = [
    {
      url: '/1',
      title: 'menuTitle1',
      icon: 'home',
      permissions: ['CanSeeUsers'],
      hidden: false
    },
    {
      url: '/2',
      title: 'menuTitle2',
      icon: 'home',
      permissions: ['CanSeeUsers'],
      hidden: false
    },
    {
      url: '/3',
      title: 'menuTitle3',
      icon: 'home',
      permissions: ['CanSeeUsers'],
      hidden: false
    },
  ];

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        NoopAnimationsModule,
        MaterialModule,
        RouterTestingModule
      ],
      declarations: [SidebarComponent]
    })
    .overrideComponent(SidebarComponent, {
      set: { changeDetection: ChangeDetectionStrategy.Default }
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SidebarComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot with menu items if expanded', () => {
    component.isExpanded = true;
    component.menuItems = menuItems;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot with menu items if collapsed', () => {
    component.isExpanded = false;
    component.menuItems = menuItems;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
