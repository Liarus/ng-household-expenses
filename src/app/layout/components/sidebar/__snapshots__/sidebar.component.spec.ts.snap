// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarComponent should match snapshot with menu items if collapsed 1`] = `
<app-sidebar
  isExpanded="false"
  menuItems={[Function Array]}
>
  <mat-sidenav-container
    class="mat-drawer-container mat-sidenav-container"
    fxflexfill=""
    fxlayout="column"
  >
    
    <div
      class="mat-drawer-backdrop ng-star-inserted"
    />
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="mat-drawer mat-sidenav ng-tns-c9-0 ng-trigger ng-trigger-transform mat-drawer-side ng-star-inserted"
      fxlayout="column"
      mode="side"
      ng-reflect-mode="side"
      ng-reflect-opened="true"
      style="transform:none;visibility:visible;0:transform;1:visibility;"
      tabindex="-1"
    >
      <div
        class="mat-drawer-inner-container"
      >
        <mat-nav-list
          class="app-nav-list mat-nav-list mat-list-base app-nav-list-collapsed"
          ng-reflect-klass="app-nav-list"
          ng-reflect-ng-class="app-nav-list-collapsed"
          role="navigation"
        >
          
          <a
            class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
            href="/1"
            mat-list-item=""
            ng-reflect-router-link="/1"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/1"
              />
              <mat-icon
                aria-hidden="true"
                class="mat-list-icon mat-icon material-icons"
                mat-list-icon=""
                role="img"
              >
                home
              </mat-icon>
              <div
                class="mat-list-text"
              >
                <p
                  class="mat-line"
                  matline=""
                >
                  menuTitle1
                </p>
              </div>
            </div>
          </a>
          <a
            class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
            href="/2"
            mat-list-item=""
            ng-reflect-router-link="/2"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/2"
              />
              <mat-icon
                aria-hidden="true"
                class="mat-list-icon mat-icon material-icons"
                mat-list-icon=""
                role="img"
              >
                home
              </mat-icon>
              <div
                class="mat-list-text"
              >
                <p
                  class="mat-line"
                  matline=""
                >
                  menuTitle2
                </p>
              </div>
            </div>
          </a>
          <a
            class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
            href="/3"
            mat-list-item=""
            ng-reflect-router-link="/3"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/3"
              />
              <mat-icon
                aria-hidden="true"
                class="mat-list-icon mat-icon material-icons"
                mat-list-icon=""
                role="img"
              >
                home
              </mat-icon>
              <div
                class="mat-list-text"
              >
                <p
                  class="mat-line"
                  matline=""
                >
                  menuTitle3
                </p>
              </div>
            </div>
          </a>
        </mat-nav-list>
      </div>
    </mat-sidenav>
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="app-sidenav mat-drawer-content mat-sidenav-content app-sidenav-content-collapsed"
      ng-reflect-klass="app-sidenav"
      ng-reflect-ng-class="app-sidenav-content-collapsed"
    />
    
  </mat-sidenav-container>
</app-sidebar>
`;

exports[`SidebarComponent should match snapshot with menu items if expanded 1`] = `
<app-sidebar
  isExpanded={[Function Boolean]}
  menuItems={[Function Array]}
>
  <mat-sidenav-container
    class="mat-drawer-container mat-sidenav-container"
    fxflexfill=""
    fxlayout="column"
  >
    
    <div
      class="mat-drawer-backdrop ng-star-inserted"
    />
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="mat-drawer mat-sidenav ng-tns-c5-0 ng-trigger ng-trigger-transform mat-drawer-side ng-star-inserted"
      fxlayout="column"
      mode="side"
      ng-reflect-mode="side"
      ng-reflect-opened="true"
      style="transform:none;visibility:visible;0:transform;1:visibility;"
      tabindex="-1"
    >
      <div
        class="mat-drawer-inner-container"
      >
        <mat-nav-list
          class="app-nav-list mat-nav-list mat-list-base app-nav-list-expanded"
          ng-reflect-klass="app-nav-list"
          ng-reflect-ng-class="app-nav-list-expanded "
          role="navigation"
        >
          
          <a
            class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
            href="/1"
            mat-list-item=""
            ng-reflect-router-link="/1"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/1"
              />
              <mat-icon
                aria-hidden="true"
                class="mat-list-icon mat-icon material-icons"
                mat-list-icon=""
                role="img"
              >
                home
              </mat-icon>
              <div
                class="mat-list-text"
              >
                <p
                  class="mat-line"
                  matline=""
                >
                  menuTitle1
                </p>
              </div>
            </div>
          </a>
          <a
            class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
            href="/2"
            mat-list-item=""
            ng-reflect-router-link="/2"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/2"
              />
              <mat-icon
                aria-hidden="true"
                class="mat-list-icon mat-icon material-icons"
                mat-list-icon=""
                role="img"
              >
                home
              </mat-icon>
              <div
                class="mat-list-text"
              >
                <p
                  class="mat-line"
                  matline=""
                >
                  menuTitle2
                </p>
              </div>
            </div>
          </a>
          <a
            class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
            href="/3"
            mat-list-item=""
            ng-reflect-router-link="/3"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/3"
              />
              <mat-icon
                aria-hidden="true"
                class="mat-list-icon mat-icon material-icons"
                mat-list-icon=""
                role="img"
              >
                home
              </mat-icon>
              <div
                class="mat-list-text"
              >
                <p
                  class="mat-line"
                  matline=""
                >
                  menuTitle3
                </p>
              </div>
            </div>
          </a>
        </mat-nav-list>
      </div>
    </mat-sidenav>
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="app-sidenav mat-drawer-content mat-sidenav-content app-sidenav-content-expanded"
      ng-reflect-klass="app-sidenav"
      ng-reflect-ng-class="app-sidenav-content-expanded"
    />
    
  </mat-sidenav-container>
</app-sidebar>
`;
